tests/test_stochatreat_assignment.py

```diff
@@ -339,3 +331,64 @@ def test_stochatreat_block_ids(df, misfit_strategy, stratum_cols):
         )
     else:
         assert n_unique_stratum_ids == n_unique_strata
+
+
+@pytest.mark.parametrize("stratum_cols", standard_stratum_cols)
+@pytest.mark.parametrize("misfit_strategy", ["global", "stratum"])
+def test_stochatreat_random_state(df, stratum_cols, misfit_strategy):
+    """
+    Tests that the results are the same on two consecutive calls with the same
+    random state
+    """
+    random_state = 42
+    treats = []
+    for _ in range(2):
+        treatments_i = stochatreat(
+            data=df,
+            stratum_cols=stratum_cols,
+            treats=2,
+            idx_col="id",
+            random_state=random_state,
+            misfit_strategy=misfit_strategy,
+        )
+        treats.append(treatments_i)
+    
+    pd.testing.assert_series_equal(
+        treats[0]["treat"], treats[1]["treat"]
+    )
+
+    
+@pytest.mark.parametrize("stratum_cols", standard_stratum_cols)
+@pytest.mark.parametrize("misfit_strategy", ["global", "stratum"])
+def test_stochatreat_shuffle_data(df, stratum_cols, misfit_strategy):
+    """
+    Tests that the mapping between idx_col and the assignments is the same on
+    two consecutive calls with the same random state and shuffled data points
+    """
+    random_state = 42
+    treats = []
+    for _ in range(2):
+        treatments_i = stochatreat(
+            data=df,
+            stratum_cols=stratum_cols,
+            treats=2,
+            idx_col="id",
+            random_state=random_state,
+            misfit_strategy=misfit_strategy,
+        )
+        treatments_i = treatments_i.sort_values("id")
+        treats.append(treatments_i)
+
+        df = df.sample(len(df), random_state=random_state)
```

Comment:
Does this still work if you `reset_index()` after the shuffle?

Reply:
Nope, should I test rather `np.testing.assert_equal(..values, ..values)` after sorting on "id"?

Reply:
What we care about being the correspondence between `idx_col` and `treat`. If we want `idx_col` to be the index than reset_index would break the original correspondence in the data and we shouldn't reset it anyway right?

Reply:
What ideally you'd want is to test the correspondence directly. Not sure what we're doing at the moment does that.